source/windows/clock.c

```diff
@@ -51,10 +51,16 @@ int aws_high_res_clock_get_ticks(uint64_t *timestamp) {
     LARGE_INTEGER ticks, frequency;
     /* QPC runs on sub-microsecond precision, convert to nanoseconds */
     if (QueryPerformanceFrequency(&frequency) && QueryPerformanceCounter(&ticks)) {
-        *timestamp = aws_timestamp_convert((uint64_t)ticks.QuadPart, AWS_TIMESTAMP_SECS, AWS_TIMESTAMP_MICROS, NULL) /
-                     (uint64_t)frequency.QuadPart;
+        uint64_t u64_ticks = ticks.QuadPart;
+        uint64_t u64_frequency = frequency.QuadPart;
+
+        uint64_t seconds = u64_ticks / u64_frequency;
+        uint64_t ticks_remainder = u64_ticks - seconds * u64_frequency;
+
+        *timestamp = aws_timestamp_convert(seconds, AWS_TIMESTAMP_SECS, AWS_TIMESTAMP_NANOS, NULL);
+        *timestamp +=
+            aws_timestamp_convert(ticks_remainder, AWS_TIMESTAMP_SECS, AWS_TIMESTAMP_NANOS, NULL) / u64_frequency;
```

Comment:
this isn't right. the tick_remainder is not in SECS, it's in u64_frequency

Reply:
The correct math for converting between different units is pretty tough, we have it in aws_timestamp_convert(). Should we add an alternative aws_timetamp_convert() that takes arbitrary u64 as the `convert_to` and `convert_from` and we could just call it like:
```
uint64-t timestamp = aws_timestamp_convert_any(
    u64_ticks/*timestamp*,
    u64_frequency/*convert_from*/,
    AWS_TIMESTAMP_NANOS/*convert_to*/,
    NULL/*remainder*/);
```

Reply:
ok. I see now how your math actually is correct. I didn't see that divide at the end. Although, it might not be correct if the frequency is greater than nanoseconds

still think we're better off adding a generic conversion function

Reply:
It is calculating `ticks_remainder` * 1e9 / `u64_frequency` which looks good to me. Mathematically `ticks_remainder`/ `u64_frequency` is in seconds and that can be converted from SECS to NANOS by multplying by 1e9.

Reply:
Oh hey, I could just do 

*timestamp += aws_timestamp_convert(ticks_remainder, u64_frequency, AWS_TIMESTAMP_NANOS, NULL);